### PR TITLE
Add Java 16 to build Spigot 1.17

### PR DESCRIPTION
**Pull request by xcgc**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [ ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### Changes to an existing Egg:

1. [x] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [x] Have you tested your Egg changes?
This pull request adds Java 16, which is required for Spigot 1.17

### CODE REVIEW

**Comment by Software-Noob** [comment]
It would be a better solution to compare if the version variable is latest or higher/equal to 1.17 before downloading files.

This makes it more efficient without wasting space and time. It most likely can be a package from debian.

**Comment by xcgc** [comment]
> It would be a better solution to move the java downloads into the version comparisons where you export the paths. It saves time and space when both are not required.
> 
> Also, you should future-proof the comparisons instead of hard-coding them. Currently, you check for `latest` or `1.17*`. It should check for the latest or greater than 1.17, such as bc/awk or multiple string comparisons (1.17* || 1.18*). Otherwise, this will break when 1.18 is released.

Done

File: minecraft/java/spigot/egg-spigot.json [hunk 1]
```diff
@@ -22,9 +26,9 @@
...
+            "container": "debian:buster-slim",
```
Comment (NgLoader) [inline minecraft/java/spigot/egg-spigot.json:30]: just use openjdk:16-jdk-alpine and remove from script the apk install javajdk8

Comment (Software-Noob) [reply minecraft/java/spigot/egg-spigot.json:30]: Java 8 is required for older spigot versions.

Comment (NgLoader) [reply minecraft/java/spigot/egg-spigot.json:30]: Okay, I tested it
Buildtools need java 8 to build the older versions

Comment (NgLoader) [reply minecraft/java/spigot/egg-spigot.json:30]: So, shut we add a new agg for newer spigot versions?
Or we need the check with version we are using and downloading the correct java version.

**Comment by NgLoader** [comment]
Thats looks better

**Review by NgLoader — APPROVED** [review]

**Comment by Software-Noob** [comment]
Thank you for your contribution!

**Pull request merged**